homeassistant/components/lektrico/binary_sensor.py

```diff
@@ -0,0 +1,118 @@
+
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_IDENTIFIERS,
+    ATTR_MANUFACTURER,
+    ATTR_MODEL,
+    ATTR_NAME,
+    ATTR_SW_VERSION,
+    CONF_FRIENDLY_NAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import LektricoDeviceDataUpdateCoordinator
+from .const import DOMAIN
+
+
+@dataclass
+class LektricoBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """A class that describes the Lektrico binary sensor entities."""
+
+    @classmethod
+    def get_is_on(cls, data: Any) -> bool | None:
+        """Return None."""
+        return None
+
+
+@dataclass
+class HeadlessBinarySensorEntityDescription(LektricoBinarySensorEntityDescription):
+    """A class that describes the Lektrico No Authentication binary sensor entity."""
+
+    @classmethod
+    def get_is_on(cls, data: Any) -> bool:
+        """Get the headless."""
+        return bool(data.headless)
+
+
+SENSORS: tuple[LektricoBinarySensorEntityDescription, ...] = (
+    HeadlessBinarySensorEntityDescription(
+        key="headless",
+        name="No Authentication",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Lektrico charger based on a config entry."""
+    _lektrico_device: LektricoDeviceDataUpdateCoordinator = hass.data[DOMAIN][
+        entry.entry_id
+    ]
+    sensors = [
+        LektricoBinarySensor(
+            sensor_desc,
+            _lektrico_device,
+            entry.data[CONF_FRIENDLY_NAME],
+        )
+        for sensor_desc in SENSORS
+    ]
+
+    async_add_entities(sensors, False)
+
+
+class LektricoBinarySensor(CoordinatorEntity, BinarySensorEntity):
+    """The entity class for Lektrico charging stations binary sensors."""
+
+    entity_description: LektricoBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        description: LektricoBinarySensorEntityDescription,
+        _lektrico_device: LektricoDeviceDataUpdateCoordinator,
+        friendly_name: str,
+    ) -> None:
+        """Initialize Lektrico charger."""
+        super().__init__(_lektrico_device)
+        self.friendly_name = friendly_name
+        self.serial_number = _lektrico_device.serial_number
+        self.board_revision = _lektrico_device.board_revision
+        self.entity_description = description
+
+        self._attr_name = f"{self.friendly_name} {description.name}"
+        self._attr_unique_id = f"{self.serial_number}_{description.name}"
```

Comment:
A name cannot be part of a unique ID :)
Use the key instead.

```suggestion
        self._attr_unique_id = f"{self.serial_number}_{description.key}"
```

Reply:
I changed it, thanks.